lib/co_aspects/aspects/deprecate_aspect.rb

```diff
@@ -0,0 +1,30 @@
+module CoAspects
+  module Aspects
+    # Generates a warning message saying that the method is deprecated and
+    # mentioning the new one.
+    #
+    # Examples
+    #
+    #   class MyClass
+    #     aspects_annotations!
+    #
+    #     _deprecate use: :new_method
+    #     def old_method
+    #     end
+    #
+    #     def new_method
+    #     end
+    #   end
+    #
+    #   MyClass.new.old_method
+    #   # => DEPRECATION WARNING: Target.old_method deprecated, use new_method instead.
+    class DeprecateAspect < Aspector::Base
```

Comment:
It might be possible that a method is deprecated and not replaced. Could we leave the `new:` param as optional, and add that info to the log in case it's being provided?